src/cryptography/hazmat/primitives/serialization/pkcs7.py

```diff
@@ -328,6 +333,33 @@ def _smime_enveloped_encode(data: bytes) -> bytes:
     return m.as_bytes(policy=m.policy.clone(linesep="\n", max_line_length=0))
 
 
+def _smime_enveloped_decode(data: bytes) -> bytes:
+    m = email.message_from_bytes(data)
+    if m.get_content_type() not in {
+        "application/x-pkcs7-mime",
+        "application/pkcs7-mime",
+    }:
+        raise ValueError("Not an S/MIME enveloped message")
+    return bytes(m.get_payload(decode=True))
+
+
+def _smime_remove_text_headers(data: bytes) -> bytes:
+    m = email.message_from_bytes(data)
+    headers = {key.title(): value for key, value in m.items()}
+    if "Content-Type" not in headers:
```

Comment:
Could you add a comment explaining that `message_from_byes` handles case insensitivity for headers? I didn't realize it did that so reminding me in the future would be good 😂

Reply:
Eh? The case insensitivty handling appears to come from the line above.

Reply:
Oh yes, title is doing the casing. I think I’d prefer just lowering

Reply:
I'm fine with both, I tried to stick with the standard case for these headers. `.lower()` might just be easier to catch though. Done!

Reply:
Actually, `get()` is straight-out a better way to handle this (case-insensitive, built-in). Modified accordingly.